### ISSUE TITLE
Class not found exception, missing PExp

### ISSUE THREAD

**Issue by fpoli** [fpoli/viper-silver#122]
Running regression test on ViperServer, we have the exception described in the title launched. This issue affects both Jenkins and running SBT locally. However, IntellJ runs ViperServer’s tests properly. Which hints in the direction of an environment problem rather than a code fault.

[Issue created by fabiopakk: 2019-09-03]
[Last updated on bitbucket: 2019-09-06]

[Comment created by fabiopakk: 2019-09-06]
Issue fixed in default branch of ViperServer repository.

**Comment by fpoli** [comment]
> **@arshavir** commented on 2016-03-11 13:55

Naming an axiom with the same identifier as the ```method bar``` causes the same issue. 

ParseAST.scala contains two suspicious case classes:

```Scala 
case class PDomainFunction1(idndef: PIdnDef, formalArgs: Seq[PFormalArgDecl], typ: PType, unique: Boolean) extends KiamaPositioned with Attributable
case class PAxiom1(idndef: PIdnDef, exp: PExp) extends KiamaPositioned with Attributable
```

Seems like these patterns are used when we do not yet know to which domain does the function or the axiom belong. Afterwards, proper PNodes are constructed via pattern matching, based on auxiliary ones. My hypothesis is that, at this point, the positions must be set explicitly, i.e. through ```viper.silver.parser.KiamaPositioned.setPos```.

**Comment by fpoli** [comment]
> **@arshavir** commented on 2016-03-14 17:48

Fixed in pull request #45
